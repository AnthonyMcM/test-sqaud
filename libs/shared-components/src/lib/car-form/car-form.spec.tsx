import { render, RenderResult, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import '@testing-library/jest-dom';

import CarForm from './car-form';

describe('CarForm', () => {
  let result: RenderResult;
  let mockOnSubmit: jest.Mock;

  beforeEach(() => {
    mockOnSubmit = jest.fn();
    result = render(<CarForm onSubmit={mockOnSubmit} />);
  });

  it('should render successfully', () => {
    const { baseElement } = result;
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = result;
    expect(baseElement).toMatchSnapshot();
  });

  it('should allow user to type', async () => {
    const formData = {
      make: 'BMW',
      model: '1 Series',
      reg: 'BC1 XC3',
      price: '1000',
    };

    const { getByLabelText } = result;
    const makeInput = getByLabelText('Make');
    await userEvent.type(makeInput, formData.make);

    const modelInput = getByLabelText('Model');
    await userEvent.type(modelInput, formData.model);

    const regInput = getByLabelText('Registration');
    await userEvent.type(regInput, formData.reg);

    const priceInput = getByLabelText('Price');
    await userEvent.type(priceInput, formData.price);

    await userEvent.click(result.getByText('Submit car'));

    expect(mockOnSubmit).toHaveBeenCalledWith(formData);
  });
});
