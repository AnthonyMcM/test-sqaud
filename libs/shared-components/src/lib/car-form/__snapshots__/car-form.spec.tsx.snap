// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarForm should match snapshot 1`] = `
<body>
  <div>
    <div
      class="ch-pa--3 ch-mt--4 ch-bg--white ch-ba--1 ch-rounded--lg ch-bc--white ch-shadow--md"
    >
      <h1
        class="ch-fs--5"
      >
        Add new car
      </h1>
      <form>
        <div
          class="ch-display--flex ch-flex-flow--row-wrap ch-mb--4"
        >
          <div
            class="ch-form__group ch-width--5 ch-mr--4 ch-rounded--lg "
          >
            <label
              class="ch-form__control-label"
              for="make"
            >
              Make
            </label>
            <input
              class="ch-form__control ch-rounded--lg"
              id="make"
              name="make"
              type="text"
              value=""
            />
          </div>
          <div
            class="ch-form__group ch-width--5"
          >
            <label
              class="ch-form__control-label"
              for="model"
            >
              Model
            </label>
            <input
              class="ch-form__control"
              id="model"
              name="model"
              type="text"
              value=""
            />
          </div>
          <div
            class="ch-form__group ch-width--5 ch-mr--4"
          >
            <label
              class="ch-form__control-label"
              for="reg"
            >
              Registration
            </label>
            <input
              class="ch-form__control"
              id="reg"
              name="reg"
              type="text"
              value=""
            />
          </div>
          <div
            class="ch-form__group ch-width--5"
          >
            <label
              class="ch-form__control-label"
              for="price"
            >
              Price
            </label>
            <input
              class="ch-form__control"
              id="price"
              name="price"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="ch-form__group"
        >
          <input
            class="ch-radio"
            id="large-1"
            name="large"
            type="radio"
            value="plan-1"
          />
          <label
            class="ch-radio__label ch-radio__label--large"
            for="large-1"
          >
            Service plan 1
            <small>
              From £6 per month
            </small>
          </label>
          <input
            class="ch-radio"
            id="large-2"
            name="large"
            type="radio"
            value="plan-2"
          />
          <label
            class="ch-mt--2 ch-radio__label ch-radio__label--large"
            for="large-2"
          >
            Service plan 2
            <small>
              From £8 per month
            </small>
          </label>
          <input
            class="ch-radio"
            disabled=""
            id="large-3"
            name="large"
            type="radio"
            value="plan-3"
          />
          <label
            class="ch-mt--2 ch-radio__label ch-radio__label--large"
            for="large-3"
          >
            Service plan 3
            <small>
              From £12 per month
            </small>
          </label>
        </div>
        <button
          class="ch-btn ch-btn--success"
          type="submit"
        >
          Submit car
        </button>
      </form>
    </div>
  </div>
</body>
`;
